test/transforms.test.ts

```diff
@@ -461,6 +463,31 @@ describe("stream transforms", () => {
             expect(mapper).toHaveBeenCalledTimes(0);
         });
 
+        test("timeout don't yield empty", async ({ expect }) => {
+            expect.assertions(3);
+
+            const mapper = vi.fn();
+
+            const testItems = [1, 1, 1, 1];
+            $.fromNext(async () => {
+                if (testItems.length > 0) {
+                    return testItems.shift();
+                }
+
+                return new Promise(() => {});
+            })
+                .batch({ timeout: 100, n: 10, yieldRemaining: true, yieldEmpty: false })
+                .map(mapper)
+                .exhaust();
+
+            await vi.advanceTimersByTimeAsync(100);
+            expect(mapper).toHaveBeenCalledTimes(1);
+            expect(mapper).toHaveBeenNthCalledWith(1, [1, 1, 1, 1]);
+
+            await vi.advanceTimersByTimeAsync(100);
+            expect(mapper).toHaveBeenCalledTimes(1);
```

Comment:
Is this missing a final assertion? Because that final call to `mapper` could be w/ an empty batch right?

Reply:
It's a bit misleading, but `toHaveBeenCalledTimes` is for the entire test, not the count since it was last checked. What I'm effectively doing here is ensuring that it wasn't called a second time, so asserting the parameters would be the same assertion as above.